Assets/MRTK/Providers/LeapMotion/LeapMotionDeviceManager.cs

```diff
@@ -0,0 +1,258 @@
+
+        /// <summary>
+        /// The LeapServiceProvider is added to the scene at runtime in OnEnable. 
+        /// </summary>
+        public LeapServiceProvider LeapMotionServiceProvider { get; protected set; }
+
+        /// <summary>
+        /// The Leap attachment hands, used to determine which hand is currently tracked by leap.
+        /// </summary>
+        public AttachmentHands LeapAttachmentHands { get; protected set; }
+
+        /// <summary>
+        /// List of hands that are currently in frame and detected by the leap motion controller. If there are no hands in the current frame, this list will be empty.
+        /// </summary>
+        public List<Hand> CurrentHandsDetectedByLeap => LeapMotionServiceProvider.CurrentFrame.Hands;
+
+        // This value can only be set in the profile, the default is LeapControllerOrientation.Headset.
+        private LeapControllerOrientation leapControllerOrientation => SettingsProfile.LeapControllerOrientation;
+
+        /// <summary>
+        /// Adds an offset to the game object with LeapServiceProvider attached.  This offset is only applied if the leapControllerOrientation
+        /// is LeapControllerOrientation.Desk and is necessary for the hand to appear in front of the main camera. If the leap controller is on the 
+        /// desk, the LeapServiceProvider is added to the scene instead of the LeapXRServiceProvider. The anchor point for the position of the leap hands is 
+        /// the position of the game object with the LeapServiceProvider attached.
+        /// </summary>
+        private Vector3 leapHandsOffset => SettingsProfile.LeapControllerOffset;
+
+        /// <summary>
+        /// Dictionary to capture all active leap motion hands detected.
+        /// </summary>
+        private readonly Dictionary<Handedness, LeapMotionArticulatedHand> trackedHands = new Dictionary<Handedness, LeapMotionArticulatedHand>();
+
+        private AttachmentHand leftAttachmentHand = null;
+        private AttachmentHand rightAttachmentHand = null;
+
+        private static readonly ProfilerMarker UpdatePerfMarker = new ProfilerMarker("[MRTK] LeapMotionDeviceManager.Update");
+
+        /// <inheritdoc />
+        public override void Enable()
+        {
+            base.Enable();
+
+            if (leapControllerOrientation == LeapControllerOrientation.Headset)
+            {
+                // If the leap controller is mounted on a headset then add the LeapXRServiceProvider to the scene
+                // The LeapXRServiceProvider can only be attached to a camera 
+                LeapMotionServiceProvider = CameraCache.Main.gameObject.AddComponent<LeapXRServiceProvider>();
+            }
+
+            if (leapControllerOrientation == LeapControllerOrientation.Desk)
+            {
+                // Create a separate gameobject if the leap controller is on the desk
+                GameObject leapProvider = new GameObject("LeapProvider");
```

Comment:
Note that if we're gonna be creating game objects in enable, we should also handle disable and destroy the same ones.

Otherwise starting/stopping services (which I think are still not fully supported but ideally would be soon) would keep creating new objects.

Reply:
Same for any other objects created by this thing.

Reply:
Included destroying the gameObjects created (AttachmentHands and LeapServiceProvider) in `Disable()`